isis/tests/FunctionalTestsGetsn.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+#include <sstream>
+
+#include <QIODevice>
+#include <QTextStream>
+#include <QStringList>
+#include <QFile>
+
+#include "getsn.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Cube.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/getsn.xml").expanded();
+static QString FROM_PARAM = "FROM=data/defaultImage/defaultCube.pvl";
+
+// check for all correct outputs
+TEST_F(DefaultCube, FunctionalTestGetsnAllTrue) {
+  QString expectedFilename = "data/defaultImage/defaultCube.pvl";
+  QString expectedSN = "Viking1/VISB/33322515";
+  QString expectedON = "Viking1/VISB/33322515";
+  QVector<QString> args = {FROM_PARAM,
```

Comment:
Since we're passing the testCube directly to getsn, do we still need to set the FROM_PARAM?